src/drive/web/modules/upload/index.js

```diff
@@ -317,23 +347,28 @@ export const uploadFilesFromNative = (
     type: ADD_TO_UPLOAD_QUEUE,
     files: files
   })
-  //!TODO Promise.All
-  for (let i = 0; i < files.length; i++) {
-    await doUpload(
-      files[i].file.fileUrl,
-      null,
-      files[i].file.name,
-      folderId,
-      'rename',
-      files[i].file.type
+
+  const promises = []
+  for (const file of files) {
+    promises.push(
+      uploadFileAndRemoveFromUploadQueue({ dispatch, file, folderId })
     )
-    dispatch(removeFileToUploadQueue(files[i].file))
   }
+
+  try {
+    await Promise.all(promises)
```

Comment:
I just asked to let the comment in order to have an idea of what we can optimize later. Here, we'll try to upload all the files in //. It seems we already have an hard way to get it working sequentially, we should not try to add parallelization for now.